source/big_core/global_defines.txt

```diff
@@ -0,0 +1,4 @@
+I_MEM_SIZE         'h2000
+I_MEM_OFFSET       'h0
+D_MEM_SIZE         'h3000
+D_MEM_OFFSET       'h2000
```

Comment:
We need to differentiate between the D_MEM size & the memory section allocated for the compiler use. (stack, heap, rom, etc)

D_MEM_GCC_SIZE - this one will be used for the linker.
D_MEM_SIZE - this one will be used in the RTL HW. (which will be used to generate the actual sram of D_MEM memory.

The (D_MEM_SIZE - D_MEM_GCC_SIZE) is the proprietary "D_MEM_SCRATCH" memory 
size "D_SCRATCH_SIZE"
And the offset of "D_SCRATCH_OFFSET" is "D_MEM_GCC_SIZE +D_MEM_OFFSET"

D_MEM_OFFSET       'h2000
D_MEM_SIZE            'h3000 //used for generating sram
D_MEM_GCC_SIZE    'h2000 // used for linker -Stack, heap, ROM
D_SCRATCH_SIZE      'h1000 (D_MEM_SIZE-D_MEM_GCC_SIZE )//((volatile int *) 
D_SCRATCH_OFFSET 'h4000 (D_MEM_OFFSET + D_MEM_GCC_SIZE)

Reply:
You helped me make those definitions clearer

Reply:
![image](https://user-images.githubusercontent.com/81047407/227804999-c79ab09a-8816-4a8a-a693-fc636fd5e794.png)
@danielk532  see this